CharacterGenerator/CharacterGenerator/CharacterTableViewController.m

```diff
@@ -0,0 +1,144 @@
+@property (weak, nonatomic) IBOutlet UITextField *nameTextField;
+@property (nonatomic, strong) IBOutletCollection(UIButton) NSArray *locationQuestionButtons;
+@property (nonatomic, strong) IBOutletCollection(UIButton) NSArray *textureQuestionButtons;
+@property (nonatomic, strong) IBOutletCollection(UIButton) NSArray *colorQuestionButtons;
+
+@property (nonatomic) NSString *location;
+@property (nonatomic) NSString *color;
+@property (nonatomic) NSString *texture;
+
+
+@end
+
+@implementation CharacterTableViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc]
+                                   initWithTarget:self
+                                   action:@selector(dismissKeyboard)];
+    [self.view addGestureRecognizer:tap];
+    
+}
+
+-(BOOL)fieldsComplete{
+    if(self.nameTextField.text.length == 0){
+        return NO;
+    }
+    for(int i=0; i < self.locationQuestionButtons.count; i++){
+        if(![self.locationQuestionButtons[i] isEnabled]){
+            break;
+        }
+        else if(i == self.locationQuestionButtons.count-1){
+            return NO;
+        }
+    }
+    for(int i=0;i<self.colorQuestionButtons.count;i++){
+        if(![self.colorQuestionButtons[i] isEnabled]){
+            break;
+        }
+        else if(i==self.colorQuestionButtons.count-1){
+            return NO;
+        }
+    }
+    for(int i=0;i<self.textureQuestionButtons.count;i++){
+        if(![self.textureQuestionButtons[i] isEnabled]){
+            break;
+        }
+        else if(i==self.textureQuestionButtons.count-1){
+            return NO;
+        }
+    }
+    return YES;
+}
+
+- (IBAction)doneButtonTapped:(UIBarButtonItem *)sender {
+    if([self fieldsComplete] == NO){
+   
+        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Missing some Field Inputs!"
+                                                        message:@"Input Data bruh"
+                                                       delegate:self
+                                              cancelButtonTitle:@"OK"
+                                              otherButtonTitles:nil];
+        [alert show];
+    }
+    else{
+        Bender *newBender = [[Bender alloc] init];
+        
+        newBender.name = self.nameTextField.text;
+        newBender.color = self.color;
+        newBender.texture = self.texture;
+        newBender.location = [newBender imageForName:self.location];
+        
+        [newBender animalForColor:self.color AndTexture:self.texture];
+        
+        
+        
+        [[BenderManager sharedManager] addBenders:newBender];
+        
+        [self dismissViewControllerAnimated:YES completion:nil];
+    }
+}
+
+- (IBAction)locationButtonTapped:(UIButton *)sender {
+    NSInteger idx = [self.locationQuestionButtons indexOfObject:sender];
+    
+    self.location = [Bender locations][idx];
+
+    // disable all other buttons
```

Comment:
Is this comment accurate? It looks like you're only disabling one of the buttons.